### PR TITLE
Build macos-arm64 pypi packages with cibuildwheel

### PR DESCRIPTION
**Pull request by joaander**
## Description

Enable `macos-arm64` builds with cibuildwheel.

## Motivation and Context
Provide `macos-arm64` builds on pypi. This is a followup to #950, see #950 for more detailed motivation.

## How Has This Been Tested?

I downloaded the build artifact, installed it, imported freud, and ran the tests:
```
▶ uname -a
Darwin cheme-kaline.local 20.6.0 Darwin Kernel Version 20.6.0: Tue Feb 22 21:10:42 PST 2022; root:xnu-7195.141.26~1/RELEASE_ARM64_T8101 arm64
▶ python
Python 3.9.12 (main, Mar 26 2022, 15:45:34) 
[Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import freud
>>> 
```

```
▶ python -m pytest .                                                                   [cibuildwheel-arm64]
==================================================================== test session starts ====================================================================
platform darwin -- Python 3.9.12, pytest-7.0.1, pluggy-1.0.0
rootdir: /Users/joaander/devel/freud, configfile: setup.cfg
plugins: anyio-3.5.0
collected 504 items                                                                                                                                         

test_box_Box.py ......................................................                                                                                [ 10%]
test_cluster.py ........                                                                                                                              [ 12%]
test_data.py ...........                                                                                                                              [ 14%]
test_density_CorrelationFunction.py .......s.......                                                                                                   [ 17%]
test_density_GaussianDensity.py ........                                                                                                              [ 19%]
test_density_LocalDensity.py ......                                                                                                                   [ 20%]
test_density_RDF.py ..........                                                                                                                        [ 22%]
test_density_SphereVoxelization.py .....                                                                                                              [ 23%]
test_diffraction_DiffractionPattern.py .............                                                                                                  [ 25%]
test_diffraction_StaticStructureFactor.py ...................................................................................s..s..s..s.....          [ 45%]
test_doctests.py ..........................                                                                                                           [ 50%]
test_environment_AngularSeparation.py .......                                                                                                         [ 51%]
test_environment_BondOrder.py ....                                                                                                                    [ 52%]
test_environment_LocalBondProjection.py ....                                                                                                          [ 53%]
test_environment_LocalDescriptors.py ............                                                                                                     [ 55%]
test_environment_MatchEnv.py ..........                                                                                                               [ 57%]
test_interface.py ....                                                                                                                                [ 58%]
test_locality_NeighborList.py .................                                                                                                       [ 61%]
test_locality_NeighborQuery.py ....................................................                                                                   [ 72%]
test_locality_PeriodicBuffer.py .......                                                                                                               [ 73%]
test_locality_Voronoi.py .........                                                                                                                    [ 75%]
test_msd_MSD.py ...                                                                                                                                   [ 75%]
test_order_Cubatic.py ....                                                                                                                            [ 76%]
test_order_Hexatic.py ............                                                                                                                    [ 79%]
test_order_Nematic.py ...                                                                                                                             [ 79%]
test_order_RotationalAutocorrelation.py ....                                                                                                          [ 80%]
test_order_SolidLiquid.py .....                                                                                                                       [ 81%]
test_order_Steinhardt.py ...................                                                                                                          [ 85%]
test_order_Translational.py ..                                                                                                                        [ 85%]
test_parallel.py ..                                                                                                                                   [ 86%]
test_pmft.py .............................................                                                                                            [ 95%]
test_util.py .......                                                                                                                                  [ 96%]
integration/test_reader_integrations.py ss...s                                                                                                        [ 97%]
validation/test_Minkowski_Structure_Metrics.py ....                                                                                                   [ 98%]
validation/test_Steinhardt_Average.py ........                                                                                                        [100%]

===================================================================== warnings summary ======================================================================
tests/integration/test_reader_integrations.py: 10 warnings
tests/validation/test_Minkowski_Structure_Metrics.py: 4 warnings
  /Users/joaander/venv/lib/python3.9/site-packages/gsd/hoomd.py:844: DeprecationWarning: Deprecated, trajectory[idx]
    warnings.warn("Deprecated, trajectory[idx]", DeprecationWarning)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================= 496 passed, 8 skipped, 14 warnings in 27.47s ========================================================
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

**Comment by joaander** [comment]
There may be some small merge conflicts with #951. I can resolve if needed.

Tommy, do I need to update the change log on this PR or do you do that later?

**Comment by tommy-waltmann** [comment]
yes, add the changelog entry on this PR.

File: .github/workflows/cibuildwheel-before-build.sh [hunk 1]
```diff
@@ -19,11 +19,23 @@ TBB_ZIP="v${TBB_VERSION}.zip"
...
+rm -rf "${PACKAGE_DIR}/tbb"
+
 # Move to a hard-coded path (defined by CIBW_ENVIRONMENT)
 mv "oneTBB-${TBB_VERSION}" "${PACKAGE_DIR}/tbb"
 cd "${PACKAGE_DIR}/tbb"
 mkdir -p build
 cd build
-cmake ../ -DTBB_TEST=OFF -DTBB_STRICT=OFF
-cmake --build . -j
+cmake ../ -DTBB_TEST=OFF -DTBB_STRICT=OFF -DCMAKE_BUILD_TYPE=Release ${EXTRA_CMAKE_ARGS}
```
Comment (tommy-waltmann) [inline .github/workflows/cibuildwheel-before-build.sh:39]: I guess its good to be explicit about having a release build, but the default build type for TBB is release, isn't it?

Comment (joaander) [reply .github/workflows/cibuildwheel-before-build.sh:39]: I saw `RelWithDebInfo` in my tests. There is no need to ship TBB's debug symbols out to all users of freud.

File: .github/workflows/cibuildwheel-before-build.sh [hunk 2]
```diff
@@ -19,11 +19,23 @@ TBB_ZIP="v${TBB_VERSION}.zip"
 curl -L -O "https://github.com/oneapi-src/oneTBB/archive/refs/tags/${TBB_ZIP}"
 unzip -q "${TBB_ZIP}"
 
+#
+EXTRA_CMAKE_ARGS=""
+if [ "${PLATFORM}" = "macos" ]; then
+    if [[ ${ARCHFLAGS} == *"arm64"* ]]; then
```
Comment (bdice) [inline .github/workflows/cibuildwheel-before-build.sh]: It would be nice to be consistent between using `[` and `[[`.

**Review by bdice — APPROVED** [review]
Thanks for this! Code changes look good. I can't test this myself so I trust the packages are good.

File: .github/workflows/cibuildwheel-before-build.sh (hunk 1 above)
Comment (tommy-waltmann) [reply .github/workflows/cibuildwheel-before-build.sh:39]: Definitely!

**Review by tommy-waltmann — APPROVED** [review]

**Pull request merged**